lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -28,14 +34,114 @@ function dosomething_northstar_menu() {
  */
 function dosomething_northstar_libraries_info() {
   $libraries['guzzle-php'] = array(
-    'name' => 'Guzzle',
-    'path' => 'lib',
+    'name' => 'guzzle',
+    'path' => 'vendor',
     'files' => array(
       'php' => array(
-        'autoload.php',
+        'autoload.php'
       ),
     ),
     'version' => 5.0,
   );
   return $libraries;
 }
+
+function dosomething_northstar_register_user($form_state) {
+  global $user;
+  // Build a user that Northstar expects.
+  $ns_user = dosomething_northstar_build_ns_user($user, $form_state);
+  // Send that to NS.
+  $client = _dosomething_northstar_build_http_client();
+  $response = drupal_http_request($client['base_url'] . '/users', array(
+    'headers' => $client['headers'],
+    'method' => 'POST',
+    'data' => json_encode($ns_user),
+    ));
+}
+
+/**
+ * Build a user json object, that's accepted by Northstar.
+ *
+ * @param obj $user
+ *  Drupal user object
+ *
+ */
+function dosomething_northstar_build_ns_user($user, $form_state) {
+  return $ns_user = array(
+    'email'         => $user->mail,
+    'mobile'        => $user->field_mobile[LANGUAGE_NONE][0]['value'],
+    'first_name'    => $user->field_first_name[LANGUAGE_NONE][0]['value'],
+    'birthdate'     => $user->field_birthdate[LANGUAGE_NONE][0]['value'],
+    'drupal_id'     => $user->uid,
+    'password'      => $form_state['values']['pass'],
+    // Address
+    'country'       => $user->field_address[LANGUAGE_NONE][0]['country'],
+    'addr_street1'  => $user->field_address[LANGUAGE_NONE][0]['thoroughfare'],
+    'addr_street2'  => $user->field_address[LANGUAGE_NONE][0]['premise'],
+    'addr_city'     => $user->field_address[LANGUAGE_NONE][0]['locality'],
+    'addr_state'    => $user->field_address[LANGUAGE_NONE][0]['administrative_area'],
+    'addr_zip'      => $user->field_address[LANGUAGE_NONE][0]['postal_code'],
+  );
+}
+
+/**
+ * Build the drupal_http_request object for nothstar calls.
+ * This will be depricated once guzzle is introduced.
+ */
+function _dosomething_northstar_build_http_client() {
+  $base_url = NORTHSTAR_URL;
+  if (getenv('DS_ENVIRONMENT') === 'local') {
+      $base_url .=  ":" . NORTHSTAR_PORT;
+    }
+  $base_url .= '/' . NORTHSTAR_VERSION;
+
+  $client = array(
+    'base_url' => $base_url,
+    'headers' => array(
+      'X-DS-Application-Id' => NORTHSTAR_APP_ID,
+      'X-DS-REST-API-Key' => NORTHSTAR_APP_KEY,
+      'Content-Type' => 'application/json',
+      'Accept' => 'application/json',
+      ),
+    );
+
+  return $client;
+}
+//@TODO: use this once we can use guzzle.
+// class Northstar {
```

Comment:
Did you want to include this commented out code or just add it after guzzle is avail?